scripts/nodes/setup-ros-node.bash

```diff
@@ -0,0 +1,340 @@
+  sed -i "s/TEMPLATES__NODES__VISIBILITY/${PKG_NAME^^}__VISIBILITY/g" $SED_FILE # visibility defines
+  sed -i "s/dummy_package_namespace/${PKG_NAME}/g" $SED_FILE # package name for includes
+  sed -i "s/dummy_node/${FILE_NAME}/g" $SED_FILE # file name
+  sed -i "s/DUMMY_NODE/${FILE_NAME^^}/g" $SED_FILE # file name for include guard
+  sed -i "s/DummyClassName/${CLASS_NAME}/g" $SED_FILE # class name
+  sed -i "s/DummyNode/${CLASS_NAME}/g" $SED_FILE # class name
+done
+
+# CMakeLists.txt: Remove comments if there any and add library
+DEL_STRINGS=("# uncomment the following" "# further" "# find_package(<dependency>")
+
+for DEL_STR in "${DEL_STRINGS[@]}"; do
+  sed -i "/$DEL_STR/d" CMakeLists.txt
+done
+
+TMP_FILE=".f_tmp"
+touch $TMP_FILE
+
+# manage dependencies here
+if [[ "$package_configured" == "no" ]]; then
+
+  # Add `THIS_PACKAGE_INCLUDE_DEPENDS` structure to simplify the rest of the file
+  TEST_LINE=`awk '$1 == "find_package(ament_cmake" { print NR }' CMakeLists.txt`  # get line before `ament_cmake` dependency
+  let CUT_LINE=$TEST_LINE-1
+  head -$CUT_LINE CMakeLists.txt >> $TMP_FILE
+
+  echo "set(THIS_PACKAGE_INCLUDE_DEPENDS" >> $TMP_FILE
+  echo "  generate_parameter_library" >> $TMP_FILE
+  echo "  rclcpp" >> $TMP_FILE
+  for DEP_PKG in "${DEP_PKGS[@]}"; do
+    if $(grep -q $DEP_PKG CMakeLists.txt); then
+      echo "'$DEP_PKG' is already listed in CMakeLists.txt"
+    else
+      echo "  $DEP_PKG" >> $TMP_FILE
+    fi
+  done
+  echo ")" >> $TMP_FILE
+
+  echo "" >> $TMP_FILE
+  echo "find_package(ament_cmake REQUIRED)" >> $TMP_FILE
+  echo "find_package(generate_parameter_library REQUIRED)" >> $TMP_FILE
+  echo "foreach(Dependency IN ITEMS \${THIS_PACKAGE_INCLUDE_DEPENDS})" >> $TMP_FILE
+  echo "  find_package(\${Dependency} REQUIRED)" >> $TMP_FILE
+  echo "endforeach()" >> $TMP_FILE
+  echo "" >> $TMP_FILE
+fi
+
+# Get line with if(BUILD_TESTING)
+TEST_LINE=`awk '$1 == "if(BUILD_TESTING)" { print NR }' CMakeLists.txt`
+let CUT_LINE=$TEST_LINE-1
+
+# Add Plugin library related stuff
+echo "# Add ${FILE_NAME} library related compile commands" >> $TMP_FILE
+echo "generate_parameter_library(${FILE_NAME}_parameters" >> $TMP_FILE
+echo "  ${NODE_PARAMS_YAML}" >> $TMP_FILE
+echo ")" >> $TMP_FILE
+
+echo "add_executable($FILE_NAME $NODE_CPP)" >> $TMP_FILE
+echo "target_include_directories($FILE_NAME PRIVATE" >> $TMP_FILE
+echo '  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"' >> $TMP_FILE
+echo '  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")' >> $TMP_FILE
+echo "target_link_libraries($FILE_NAME ${FILE_NAME}_parameters)" >> $TMP_FILE
+echo "ament_target_dependencies($FILE_NAME \${THIS_PACKAGE_INCLUDE_DEPENDS})" >> $TMP_FILE
+
+if [[ "$package_configured" == "no" ]]; then
+
+  ## Add install directives
+  echo "" >> $TMP_FILE
+  echo "install(" >> $TMP_FILE
+  echo "  TARGETS" >> $TMP_FILE
+  echo "  $FILE_NAME" >> $TMP_FILE
+  echo "  RUNTIME DESTINATION lib/${PROJECT_NAME}" >> $TMP_FILE
```

Comment:
single quotes are required here
```suggestion
  echo '  RUNTIME DESTINATION lib/${PROJECT_NAME}' >> $TMP_FILE
```

Reply:
why? I want that ${PROJECT_NAME} is parsed to the string. so this should work